### ISSUE TITLE
Not working.

### ISSUE THREAD

**Issue by Faltfromoss** [centic9/poi-on-android#16]
You wrote:

> Getting started
> Use a pre-built jar
> If you want to get started quickly, there is a ready-made jar-file available in the release section.
> You should be able to simply add this to your Android project and use the Apache POI classes from it.

I added the "poisahdow-all,jar" file to an empty test project. And I tried to just refer to the class from this library:

```
public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView message = (TextView)findViewById(R.id.messageTV);
        try {
            Workbook wb = new XSSFWorkbook();
            message.setText("The book instance was created.");
        }
        catch (Exception ex){
            message.setText(ex.getMessage());
        }        
    }
}
```
The contents of the "build.gradle" file are standard with the option enabled `multiDexEnabled = ture`:
When I try to run this application, the compilation runs successfully. But when you run this application and when you try to create an instance of a `WorkBook` in this row:
`try {
            Workbook wb = new XSSFWorkbook();
        }`
That application is closed with the message:
![10](https://user-images.githubusercontent.com/18700275/28243679-c555c266-69dc-11e7-8e0e-c466b399b993.png)
And the exception is not intercepted by try-catch block and it is unclear for what reason this occurs.
Tell me please, what am I doing wrong? Why does not this library want to work on android?

**Comment by centic9** [comment]
Is there anything in the logs? If you are connected via adb, you should be able to view them, they might contain details.

**Comment by Faltfromoss** [comment]
Here's what I got
![11](https://user-images.githubusercontent.com/18700275/28247853-060ac12a-6a42-11e7-80af-8e115d0dd2cb.png)

**Comment by centic9** [comment]
Please set the necessary system properties as explained in the README, I have moved the instructions up in the file to make them easier to spot.

**Comment by Faltfromoss** [comment]
I have next problem

**Comment by Faltfromoss** [comment]
Ok. I apologize for the inattention. I added these settings and everything worked. But now there was another problem. I tried to open a file that I need to edit with the InputStream. The file size is ~1mb. 
```
File file = getExternalPath();   // here I get target file    
        if(!file.exists()){
            messageTextView.setText("File not found");
            return;
        }
        try (InputStream inp = new FileInputStream(file))
         {
            Workbook wb = WorkbookFactory.create(inp);
            messageTextView.setText(String.valueOf(wb.getNumberOfSheets()));
        } catch (IOException ex) {
            messageTextView.setText(ex.getMessage());
       }
```
And it works, but the function `wb.getNumberOfSheets()`  is processed 30-50 seconds! That is, the application for this time just hangs. And in general, whenever you turn to the book, there is such a hang. However, if you refer to an empty excel file, the function executes instantly. The [documentation "Apache poi"](https://poi.apache.org/spreadsheet/quick-guide.html#FileInputStream) states that the use of InputStream requires more memory than using the file. I tried to connect to the book using a File.

`Workbook wb = WorkbookFactory.create(file);`

But thus I receive exception:

`org.apache.poi.openxml4j.util.ZipSecureFile$ThresholdInputStream cannot be cast to java.util.zip.ZipFile$RAFStream`

I can not understand what the problem is again :/
